edx/edx-ora2
# Jkantor/pylint

**TL;DR -** Pylint was silently failing

JIRA: [AU-26](https://openedx.atlassian.net/browse/AU-26)

**What changed?**

- this PR (https://github.com/edx/edx-ora2/pull/1594) added the --django-settings-module flag to the pylint call, but pylint fails because `edx-ora2.settings` isn't a python module. pylint would explode and write nothing to the expected output file, and the post-pylint output file processing script we had interpreted a blank file as a success with 0 failures.
- removing the `edx-ora2` from the module name makes it point to the settings folder, which is a valid python module
- also added code to `run-pylint.sh` to check if pylint exploded, and if so, exit with an error code so travis knows something went wrong.
- I also added the acceptance tests to pylint ignore, because they've been nonfunctional, untouched, and unused for a while now. Doing so resulted in big changes to `pylintrc`, but it's mostly edx-lint reorganizing things alphabetically.

- **Note**: The majority of the lint errors are something i believe is a newly enforced rule in edx-lint: `logging-format-interpolation`. This rule wants you to use old-style string formatting (`'format var = %s' % param`) and pass the params as args to the logging function rather than string.format(). I'm not entirely sure why, but the logging documentation insists that this is best practice. Something about lazy evaluation I think. 

**Developer Checklist**


**Testing Instructions**

Look at the quality test for this PR, and make sure the call to pylint doesn't explode.
You can also checkout this branch and run `make quality` in your virtualenv and make sure pylint doesn't fail to run

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FYI: @edx/masters-devs-gta
